src/views/ApplicationPage/components/ServiceHistoryListItem.tsx

```diff
@@ -0,0 +1,236 @@
+}
+
+const ServiceHistoryListItem = ({ idx, serviceHistoryList, setServiceHistoryList }: ServiceHistoryListItem) => {
+  const [isServiceClicked, setIsServiceClicked] = useState(false);
+  const [isPeriodClicked, setIsPeriodClicked] = useState(false);
+
+  const activateServiceBox = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    isServiceClicked ? setIsServiceClicked(false) : setIsServiceClicked(true);
+
+    const tempService = event.currentTarget.innerText;
+    const tempServiceHistoryList = serviceHistoryList.map((item, i) => {
+      if (i === idx) {
+        return {
+          ...item,
+          service: tempService,
+        };
+      }
+      return item;
+    });
+
+    setServiceHistoryList(tempServiceHistoryList);
+  };
+
+  const activatePeriodBox = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    isPeriodClicked ? setIsPeriodClicked(false) : setIsPeriodClicked(true);
+
+    const tempPeriod = event.currentTarget.innerText;
+    const tempServiceHistoryList = serviceHistoryList.map((item, i) => {
+      if (i === idx) {
+        return {
+          ...item,
+          period: tempPeriod,
+        };
+      }
+      return item;
+    });
+
+    setServiceHistoryList(tempServiceHistoryList);
+  };
+
+  const deleteHistory = () => {
+    const tempServiceHistoryList = serviceHistoryList.filter((_, i) => i !== idx);
+    if (tempServiceHistoryList.length >= 0) {
+      setServiceHistoryList(tempServiceHistoryList);
+    }
+  };
+
+  return (
+    <S.ServiceHistoryListItemLayout>
+      <S.SelectBox $height={idx}>
+        <S.SelectServiceBox
+          $isServiceClicked={isServiceClicked}
+          onClick={() => {
+            isServiceClicked ? setIsServiceClicked(false) : setIsServiceClicked(true);
+          }}>
+          <input type="button" value={serviceHistoryList[idx].service} />
+          {isServiceClicked ? <IcUpBlue /> : <IcDownGrey />}
+        </S.SelectServiceBox>
+        <div>
+          {isServiceClicked && (
+            <S.SelectDetailList>
+              <li>
+                <button type="button" onClick={activateServiceBox}>
+                  펌
+                </button>
+              </li>
+              <li>
+                <button type="button" onClick={activateServiceBox}>
+                  탈색
+                </button>
+              </li>
+              <li>
+                <button type="button" onClick={activateServiceBox}>
+                  블랙 염색
+                </button>
+              </li>
+              <li>
+                <button type="button" onClick={activateServiceBox}>
+                  컬러 염색
+                </button>
+              </li>
+            </S.SelectDetailList>
+          )}
+        </div>
+      </S.SelectBox>
+      <S.SelectBox $height={idx}>
+        <S.SelectPeriodBox
+          $isPeriodClicked={isPeriodClicked}
+          onClick={() => {
+            isPeriodClicked ? setIsPeriodClicked(false) : setIsPeriodClicked(true);
```

Comment:
```suggestion
             setIsServiceClicked((prev) => !prev)
```
이렇게 하면 간단하게 값을 바꿔줄 수 있지롱 ~~~
setState 함수 사용 시 이전 state 값을 사용하고 싶으면 prevState를 사용하면 됩니당!